### ISSUE TITLE
[BUG] SMTP setup 

### ISSUE THREAD

**Issue by cmathews393** [stumpapp/stump#429]
**Describe the bug**
Unable to create SMTP mailer in settings:

**To Reproduce**

1. Deploy from docker compose
2. Attempt to set up email in settings
3. Potentially caused by pasting 587 into the smtp port field, but same issue occurs on max attachment size without touching it so probably unrelated to pasting.

**Expected behavior**
Accept a typed value as entered.
**Screenshots**
![image](https://github.com/user-attachments/assets/36a2653a-4d0e-4db0-bcd2-3d1420af192f)

**Desktop (please complete the following information):**

- OS: [e.g. macOS, Windows, Linux, iOS, watchOS, Android]

Server running from docker compose on ubuntu, accessing from Edge on Windows 10

**Additional context**

**Comment by aaronleopold** [comment]
Hey 👋 

This looks to just be a client-side validation issue, so the fix should hopefully be quick. I'll try to give it a look before the weekend ends, but I have a couple other items in the queue first. I'll post an update when I put up a fix, thanks for making the report!

**Comment by aaronleopold** [comment]
OK scratch that - I had some time while my dinner is in the oven 🥘 

I think I've fixed the issue locally, I'll make sure the fix goes to `nightly` sometime tomorrow. I'm taking this as an opportunity to start adding proper testing on the frontend application so little issues like this are less likely to surface as the app continues to get bigger and bigger.

I also noticed the delete functionality was never mapped to the delete button? lol so I quickly added that too

<details>
<summary>Demo</summary>

https://github.com/user-attachments/assets/47cd5911-6c46-4da2-9bb5-c2c4a3a67d2f

</details>
